src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/CostFunctionsForPhysicalPlans.java

```diff
@@ -0,0 +1,20 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+
+public interface CostFunctionsForPhysicalPlans {
+
+    int getTotalNumberOfRequests( PhysicalPlan pp ) throws QueryOptimizationException;
```

Comment:
Now I suggest to rename these methods to `determine...` (e.g., `determineTotalNumberOfRequests`) which makes more clear that they are not simply getter methods but they involve some processing.